.github/workflows/ci.yml

```diff
@@ -0,0 +1,29 @@
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ '*' ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go: ["1.20.x", "1.21.x"]
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: ${{ matrix.go }}
+        cache: true
+    - name: Download dependencies
+      run: go mod download
```

Comment:
You're right, this isn't absolutely necessary. It's there only to get a nice separate "download dependencies" step so that "run tests" timings don't include that.

RE: Makefile: Maybe? Not a bad idea. Although note that this is unrelated to the GOBIN business in the Makefile. That's for tool management, e.g. `go install example.com/golint` should not install to a global directory but to $project/bin.